### DISCUSSION TITLE
How can I get the schema when using StreamDecoder when the batches are empty?

### DISCUSSION THREAD

**Discussion by jwimberl** [apache/arrow-rs#5947] [General]
The new StreamDecoder utility reads the arrow SchemaRef from a `MessageHeader::Schema`-type message into a private struct member named `schema`; however, there isn't a public method to retrieve this schema that I can see. When the data is empty and there are no batches, is there any way to still get the decoded SchemaRef?

**Comment by Jefffrey** [comment]
Issue has been raised for this feature: https://github.com/apache/arrow-rs/issues/6420

Will close this so any further discussion can be on the issue
